core/src/main/java/com/linecorp/armeria/client/retry/Backoff.java

```diff
@@ -84,19 +85,60 @@ static Backoff exponential(long initialDelayMillis, long maxDelayMillis, double
+     * jitter value is 0.3, the ultimate backoff delay will be chosen between 1000 * (1 - 0.3)
+     * and 1000 * (1 + 0.3) by {@link ThreadLocalRandom}.
+     * The rate value should be between {@code 0.0f} and {@code 1.0f}.
+     *
+     * @param jitterRate the rate that used to calculate the backoff delay bound
+     * @throws IllegalArgumentException if {@code jitterRate} is a negative value or greater than 1.0f
+     */
+    default Backoff withJitter(double jitterRate) {
+        checkArgument(0.0f <= jitterRate && jitterRate <= 1.0f,
+                      "jitterRate: %s (expected: >= 0.0f and <= 1.0f)", jitterRate);
+        return withJitter(-jitterRate, jitterRate, ThreadLocalRandom::current);
+    }
+
+    /**
+     * Returns a {@link Backoff} that provides a jitter value for the backoff delay using
+     * <a href="https://www.awsarchitectureblog.com/2015/03/backoff.html">full jitter</a> strategy.
+     * The {@code minJitterRate} and {@code maxJitterRate} will be multiplied to backoff delay to
+     * calculate the bound. For example, if the calculated delay using {@link Backoff#exponential(long, long)}
+     * is 1000 millis and the {@code minJitterRate} is -0.2, {@code maxJitterRate} is 0.3, the ultimate
+     * backoff delay will be chosen between 1000 * (1 - 0.2) and 1000 * (1 + 0.3) by {@link ThreadLocalRandom}.
+     * The rate values should be between {@code -1.0f} and {@code 1.0f}.
+     *
+     * @param minJitterRate the rate that used to calculate the lower bound of backoff delay
+     * @param maxJitterRate the rate that used to calculate the upper bound of backoff delay
+     * @throws IllegalArgumentException if {@code minJitterRate} is greater than {@code maxJitterRate} or if the
+     *                                  {@code minJitterRate} and {@code maxJitterRate} values are not in
+     *                                  between {@code -1.0f} and {@code 1.0f}.
      */
-    default Backoff withJitter(long minJitterMillis, long maxJitterMillis) {
-        return withJitter(minJitterMillis, maxJitterMillis, ThreadLocalRandom::current);
+    default Backoff withJitter(double minJitterRate, double maxJitterRate) {
+        return withJitter(minJitterRate, maxJitterRate, ThreadLocalRandom::current);
     }
 
     /**
-     * Returns a {@link Backoff} that provides a delay that increases using
+     * Returns a {@link Backoff} that provides a jitter value for the backoff delay using
      * <a href="https://www.awsarchitectureblog.com/2015/03/backoff.html">full jitter</a> strategy.
+     * The {@code minJitterRate} and {@code maxJitterRate} will be multiplied to backoff delay to
+     * calculate the bound. For example, if the calculated delay using {@link Backoff#exponential(long, long)}
+     * is 1000 millis and the {@code minJitterRate} is -0.2, {@code maxJitterRate} is 0.3, the ultimate
+     * backoff delay will be chosen between 1000 * (1 - 0.2) and 1000 * (1 + 0.3) by {@code randomSupplier}.
```

Comment:
`by {@code randomSupplier}` -> Probably OK to remove this given that we have `@param` tag that explains it?